xapian-letor/api/letor.cc

```diff
@@ -73,13 +72,13 @@ Letor::prepare_training_file(const string & query_file, const string & qrel_file
 }
 
 void
-Letor::create_ranker(int ranker_type, int metric_type) {
-    switch(ranker_type) {
-        case 0: internal->ranker = * new SVMRanker;
-                break;
-        case 1: break;
-        default: ;//cout<<"Please specify proper ranker.";
-    }
+Letor::create_ranker(int ranker_type, int metric_type) { //TODO: update when adding rankers
```

Comment:
I'm pretty sure as we re-introduce multiple rankers, that we shouldn't have this function here at all. Client code should just create the ranker object is needs. Command line utilities would be better off having `--ranker=<name>` (which will result in code a little like this, but will generally be more clear). So it may be worth going through and eliminating this function entirely, fairly soon.